lib/client/util/__mocks__/Geolocation.js

```diff
@@ -0,0 +1,37 @@
+var path = require.requireActual('../Geolocation');
+
+var geoMock = jest.genMockFromModule('../Geolocation');
+
+var _coords = null;
+var _enabled = false;
+function __setLocation(lat, lon) {
+  _enabled = true;
+  _coords = { latitude: lat, longitude: lon };
+}
+
+function __enable() {
+  _enabled = true;
+}
+
+function __disable() {
+  _coords = null;
+  _enabled = false;
+}
+
+function getCurrentPosition(successCallback, errorCallback) {
+  if(_enabled) {
+    if(_coords) {
+      successCallback({ coords: _coords });
+    }
```

Comment:
I see now why you did it this way... to simulate a delay in getting the result, right? I wonder if some better names could be used in the mock to make it clearer the scenarios that are being setup... Maybe instead of disable, "reset"? "disable" could indicate a permission denied error, or even just name it "denyPermission". Then to simulate the loading state, you can have setLocation take a delay param and use setTimeout to delay the success callback when calling getLocation.